.github/workflows/ci-publish.yml

```diff
@@ -0,0 +1,127 @@
+          echo "<settings><servers><server><id>ossrh</id><username>$OSSRH_USERNAME</username><password>$OSSRH_PASSWORD</password></server></servers></settings>" > ~/.m2/settings.xml
+        env:
+          MASTER_PASSWORD: ${{ secrets.MASTER_PASSWORD }}
+          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+      # TODO: the step "Make JanusGraph JAR artifacts" should also create signature key for full JanusGraph bundle but for some
+      # reason it doesn't sign full distribution (only standard distribution).
+      # After this is fixed we can delete the step "Sign JanusGraph full distribution artifact".
+      - name: Make JanusGraph JAR artifacts
+        run: mvn clean install -Pjanusgraph-release -DskipTests=true
+      - name: Print content of the generated target directory for debugging purposes
+        run: echo $(ls janusgraph-dist/target/)
+      - name: Deploy JanusGraph JAR artifacts into staging repository
+        run: mvn deploy -Pjanusgraph-release -DskipTests=true
+      - name: Sign JanusGraph full distribution artifact
+        run: gpg --armor --detach-sign --yes janusgraph-dist/target/${JG_FULL_DIST_NAME}.zip
+      - name: Install requirements to build documentation
+        run: pip3 install -r requirements.txt
+      - name: Build documentation
+        run: mkdocs build
+      - name: Pack documentation into zip
+        run: |
+          mv site ${JG_DOC_DIST_NAME}
+          zip -r ${JG_DOC_DIST_NAME}.zip ${JG_DOC_DIST_NAME}
+      - name: Sign documentation artifact
+        run: gpg --armor --detach-sign ${JG_DOC_DIST_NAME}.zip
+      - name: Check distribution signature
+        run: gpg --verify janusgraph-dist/target/${JG_DIST_NAME}.zip.asc janusgraph-dist/target/${JG_DIST_NAME}.zip
+      - name: Check full distribution signature
+        run: gpg --verify janusgraph-dist/target/${JG_FULL_DIST_NAME}.zip.asc janusgraph-dist/target/${JG_FULL_DIST_NAME}.zip
+      - name: Check documentation signature
+        run: gpg --verify ${JG_DOC_DIST_NAME}.zip.asc ${JG_DOC_DIST_NAME}.zip
+      - uses: ncipollo/release-action@v1
+        with:
+          allowUpdates: true
+          artifactErrorsFailBuild: true
+          draft: true
+          tag: "${{ env.RELEASE_TAG }}"
+          artifacts: "KEYS,${{ env.JG_DOC_DIST_NAME }}.zip.asc,${{ env.JG_DOC_DIST_NAME }}.zip,janusgraph-dist/target/${{ env.JG_DIST_NAME }}.zip.asc,janusgraph-dist/target/${{ env.JG_DIST_NAME }}.zip,janusgraph-dist/target/${{ env.JG_FULL_DIST_NAME }}.zip.asc,janusgraph-dist/target/${{ env.JG_FULL_DIST_NAME }}.zip"
+          name: "${{ env.JG_VER }}"
+          omitBodyDuringUpdate: true
+          omitNameDuringUpdate: true
+          omitPrereleaseDuringUpdate: true
+          owner: janusgraph-bot
+          prerelease: true
+          replacesArtifacts: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+          body: |
+            # Version ${{ env.JG_VER }} (MM DD, YYYY)
```

Comment:
Shouldn't it be possible to set the date automatically?

Reply:
We used the date which is set in changelog file and not the date of the draft release been created.
I.e. the date the draft release is created might be not equal to the date of the release. I don't know the easy way how to automate it at this moment

Reply:
If we find a solution, it would be great. I'm okay with it for now. We could open an issue with follow up tasks.